[rpc] Make RPC timeout APIs consistent.

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#37027 [rpc] Make RPC timeout APIs consistent.**

We now have timeout parameters in rpc_sync and async accepting float types, so we should change the types in _set_rpc_timeout, _get_rpc_timeout, and `RpcBackendOptions` to be consistent with this. Also modified the associated documentation.

Added tests to ensure that we can initialize `RpcBackendOptions` with a float for timeout.

Differential Revision: [D21125171](https://our.internmc.facebook.com/intern/diff/D21125171/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21125171/)!

rohan-varma committed:
[WIP][rpc] Move _set_rpc_backand and RpcBackendOptions to use float instead of
timedelta

The RPC timeout passed into rpc_sync and rpc_async after the below
change is now float, so we should make these APIs consistent.

Differential Revision: [D21125171](https://our.internmc.facebook.com/intern/diff/D21125171/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21125171/)!

[ghstack-poisoned]

rohan-varma committed:
Update on "[WIP][rpc] Move _set_rpc_backand and RpcBackendOptions to use float instead of
timedelta"

timedelta**
* #34650 per-RPC timeouts for rpc_sync and rpc_async

timedelta

The RPC timeout passed into rpc_sync and rpc_async after the below
change is now float, so we should make these APIs consistent.

Differential Revision: [D21125171](https://our.internmc.facebook.com/intern/diff/D21125171/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21125171/)!

[ghstack-poisoned]

rohan-varma committed:
Update on "[WIP][rpc] Move _set_rpc_backand and RpcBackendOptions to use float instead of
timedelta"

timedelta**
* #34650 per-RPC timeouts for rpc_sync and rpc_async
timedelta**
* #34650 per-RPC timeouts for rpc_sync and rpc_async

timedelta

The RPC timeout passed into rpc_sync and rpc_async after the below
change is now float, so we should make these APIs consistent.

Differential Revision: [D21125171](https://our.internmc.facebook.com/intern/diff/D21125171/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21125171/)!

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] Make RPC timeout APIs consistent. "


We now have timeout parameters in rpc_sync and async accepting float types, so we should change the types in _set_rpc_timeout, _get_rpc_timeout, and `RpcBackendOptions` to be consistent with this. Also modified the associated documentation.

Added tests to ensure that we can initialize `RpcBackendOptions` with a float for timeout.

Differential Revision: [D21125171](https://our.internmc.facebook.com/intern/diff/D21125171/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21125171/)!

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] Make RPC timeout APIs consistent. "


We now have timeout parameters in rpc_sync and async accepting float types, so we should change the types in _set_rpc_timeout, _get_rpc_timeout, and `RpcBackendOptions` to be consistent with this. Also modified the associated documentation.

Added tests to ensure that we can initialize `RpcBackendOptions` with a float for timeout.

Differential Revision: [D21125171](https://our.internmc.facebook.com/intern/diff/D21125171/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21125171/)!

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] Make RPC timeout APIs consistent. "


We now have timeout parameters in rpc_sync and async accepting float types, so we should change the types in _set_rpc_timeout, _get_rpc_timeout, and `RpcBackendOptions` to be consistent with this. Also modified the associated documentation.

Added tests to ensure that we can initialize `RpcBackendOptions` with a float for timeout.

Differential Revision: [D21125171](https://our.internmc.facebook.com/intern/diff/D21125171/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21125171/)!

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] Make RPC timeout APIs consistent. "


We now have timeout parameters in rpc_sync and async accepting float types, so we should change the types in _set_rpc_timeout, _get_rpc_timeout, and `RpcBackendOptions` to be consistent with this. Also modified the associated documentation.

Added tests to ensure that we can initialize `RpcBackendOptions` with a float for timeout.

Differential Revision: [D21125171](https://our.internmc.facebook.com/intern/diff/D21125171/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21125171/)!

[ghstack-poisoned]

mrshenli commented on torch/csrc/distributed/rpc/init.cpp:
```diff
@@ -626,15 +630,22 @@ If the future completes with an error, an exception is thrown.
       "_set_rpc_timeout",
-      [](const std::chrono::milliseconds& rpcTimeout) {
+      [](const float rpcTimeoutSeconds) {
+        const auto secToMsConversion = 1000;
```
Shall we use one const for `secToMsConversion` and `msToSecDiv`? It can be a float as we are doing the cast anyway?

rohan-varma replied:
Changed it to be a single const

mrshenli commented on torch/csrc/distributed/rpc/init.cpp:
```diff
@@ -626,15 +630,22 @@ If the future completes with an error, an exception is thrown.
-          raised. To control timeout for specific RPCs, a timeout parameter can
-          be passed into :meth:`~torch.distributed.rpc.rpc_sync` and
+          Set the default timeout for all RPCs. The input unit is expected to be
+          in seconds.If an RPC is not completed within this time, an exception
```
seconds.If -> seconds. If (space)

mrshenli commented on torch/distributed/rpc/_testing/faulty_agent_backend_registry.py:
```diff
@@ -62,7 +64,7 @@ def _faulty_process_group_init_backend_handler(
             group,
             rpc_backend_options.num_send_recv_threads,
-            rpc_backend_options.rpc_timeout,
+            timedelta(seconds=rpc_backend_options.rpc_timeout),
```
Do we plan to make this take a float too in subsequent PRs?

rohan-varma replied:
@mrshenli I decided to not change the internal representations in the RPC agents, the main purpose was to make sure that from the user perspective they pass in float whenever constructing a `RpcBackendOptions` or calling the set/get default timeout APIs. So, the change to float is only planned to be in RpcBackendOptions structs and the relevant APIs.

rohan-varma committed:
Update on "[rpc] Make RPC timeout APIs consistent. "


We now have timeout parameters in rpc_sync and async accepting float types, so we should change the types in _set_rpc_timeout, _get_rpc_timeout, and `RpcBackendOptions` to be consistent with this. Also modified the associated documentation.

Added tests to ensure that we can initialize `RpcBackendOptions` with a float for timeout.

Differential Revision: [D21125171](https://our.internmc.facebook.com/intern/diff/D21125171/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21125171/)!

[ghstack-poisoned]

jjlilley commented:
OSS LGTM, but can you double-check the FB-only Thrift side? It looks like there's a build failure

rohan-varma committed:
Update on "[rpc] Make RPC timeout APIs consistent. "


We now have timeout parameters in rpc_sync and async accepting float types, so we should change the types in _set_rpc_timeout, _get_rpc_timeout, and `RpcBackendOptions` to be consistent with this. Also modified the associated documentation.

Added tests to ensure that we can initialize `RpcBackendOptions` with a float for timeout.

Differential Revision: [D21125171](https://our.internmc.facebook.com/intern/diff/D21125171/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21125171/)!

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] Make RPC timeout APIs consistent. "


We now have timeout parameters in rpc_sync and async accepting float types, so we should change the types in _set_rpc_timeout, _get_rpc_timeout, and `RpcBackendOptions` to be consistent with this. Also modified the associated documentation.

Added tests to ensure that we can initialize `RpcBackendOptions` with a float for timeout.

Differential Revision: [D21125171](https://our.internmc.facebook.com/intern/diff/D21125171/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21125171/)!

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] Make RPC timeout APIs consistent. "


We now have timeout parameters in rpc_sync and async accepting float types, so we should change the types in _set_rpc_timeout, _get_rpc_timeout, and `RpcBackendOptions` to be consistent with this. Also modified the associated documentation.

Added tests to ensure that we can initialize `RpcBackendOptions` with a float for timeout.

Differential Revision: [D21125171](https://our.internmc.facebook.com/intern/diff/D21125171/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21125171/)!

[ghstack-poisoned]

jjlilley reviewed (approved):
Thrift fixes LGTM, thanks!

rohan-varma committed:
Update on "[rpc] Make RPC timeout APIs consistent. "


We now have timeout parameters in rpc_sync and async accepting float types, so we should change the types in _set_rpc_timeout, _get_rpc_timeout, and `RpcBackendOptions` to be consistent with this. Also modified the associated documentation.

Added tests to ensure that we can initialize `RpcBackendOptions` with a float for timeout.

Differential Revision: [D21125171](https://our.internmc.facebook.com/intern/diff/D21125171/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21125171/)!

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] Make RPC timeout APIs consistent. "


We now have timeout parameters in rpc_sync and async accepting float types, so we should change the types in _set_rpc_timeout, _get_rpc_timeout, and `RpcBackendOptions` to be consistent with this. Also modified the associated documentation.

Added tests to ensure that we can initialize `RpcBackendOptions` with a float for timeout.

Differential Revision: [D21125171](https://our.internmc.facebook.com/intern/diff/D21125171/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21125171/)!

[ghstack-poisoned]
